lib/packages/core/lib/src/utils/media/media_handler.dart

```diff
@@ -31,6 +32,12 @@ class LMChatMediaHandler {
       pickedMedia.add(media);
     } else if (media is List<LMChatMediaModel>) {
       pickedMedia.addAll(media);
+    } else if (media is LMChatAttachmentViewData) {
```

Comment:
Here we are checking `media` type for both `LMChatMediaModel` and `LMChatAttachmentViewData`. why are we adding media in 2 different format. i guess  one of the type check would be redundant.if it is true then i would suggest you to modify this function to take 2 nullable args i.e: `addPickedMedia({LMChatAttachmentViewData? media, List<LMChatAttachmentViewData>? mediaList })` and in the body of the function just check if media not null `  pickedMedia.add(media.toMediaModel())` and same for other. it would save type checking on runtime and also make this function TypeSafe.

Reply:
This is because the function is being used in multiple places, and I can't stick it to a single type either in params or returned result.

Reply:
There are 2 different problem in this function: 
### First
i am not asking to accept a single type of param in this function, currently it have to deal with 4 data type
1. LMChatMediaModel
2. List< LMChatMediaModel>
3. LMChatAttachmentViewData
4. List< LMChatAttachmentViewData>
As i can see in function itself `LMChatMediaModel` and `LMChatAttachmentViewData` are inter-convertable. that mean where we are using this function with `LMChatMediaModel` currrently can use it with `LMChatAttachmentViewData` and where we are using it with `List< LMChatMediaModel>` can use as `List< LMChatAttachmentViewData>`. and the return type of this function is anyway `void` so you do not have to deal with it.

### How to make it TypeSafe
from the fucntion body i can understand that this function help in adding mediaModel/attachmentViewData to the pickedMedia. List for multiple media and Object for single media. so instead of using dynamic we can reframe this function to something like this:
```dart
 void addPickedMedia({LMChatAttachmentViewData? singleMedia, List<LMChatAttachmentViewData>? multipleMedia}) {
if(singleMedia != null) {
 pickedMedia.add(singleMedia);
} else if (multipleMedia != null) {
pickedMedia.addAll(multipleMedia);
}
   }
```
or 
```dart
void addPickedMedia({List<LMChatAttachmentViewData> media}) {
pickedMedia.addAll(media);
   }

//and when using it for single `LMChatAttachmentViewData` 
LMChatAttachmentViewData media = LMChatAttachmentViewData();
addPickedMedia([media]);
```